opal/class/opal_fifo.h

```diff
@@ -228,7 +229,10 @@ static inline opal_list_item_t *opal_fifo_pop_atomic(opal_fifo_t *fifo)
             attempt = 0;
         }
 
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wlanguage-extension-token"
```

Comment:
I'd rather get rid of the typeof() usage in the ll/sc code.  I have a bunch of patches to clean up the atomics code; let me go see how bad that change would make things and then we can come back to this.  But we really don't want to pragma every usage of the opal_atomic_ll code.

Reply:
We only have to protect the use of opal_atomic_ll because they are defined as macros (and the diagnostic do not apply to macros when they are read, but only when they are applied. If we convert he macro into an inlined function then we can have the pragma in a single location.

Reply:
Nathan originally used inline functions, but even with always_inline primitives, couldn't keep all the compilers from sometimes not inlining the function.  And once not inlined, there was a very high probability of the link failing due to some write in setting up the function call, causing the conditional store to fail, infinitely.

But I think I found a better solution to this problem.  Unfortunately, I checked in my branch full of atomic changes.  I'll post it later today.